inc/dropdown.class.php

```diff
@@ -1478,7 +1478,8 @@ static function showNumber($myname, $options = []) {
 
       $field_id = Html::cleanId("dropdown_".$myname.$p['rand']);
       if (!isset($p['toadd'][$p['value']])) {
-         $valuename = self::getValueWithUnit($p['value'], $p['unit']);
+         $decimals = Toolbox::isFloat($p['step']) ? null : 0;
```

Comment:
I do not agree with this. The number should be formatted even if it is a float. Indeed, the decimal separator is not always a `.`.

Another solution could be to extract the number of decimals from the step value, like this:
```suggestion
         $decimals = strlen(preg_replace('/\d*\./', '', floatval($p['step'])));
```

Reply:
Done, but I've keep "0" if the number is not a float, to avoid adding unnecessary `,0000` parts to integers values.

Reply:
![image](https://user-images.githubusercontent.com/42734840/151766730-95a717eb-0c64-4bc2-aeb5-4006603b7543.png)